src/modules/resource/reslog.c

```diff
@@ -217,8 +217,7 @@ int reslog_post_pack (struct reslog *reslog,
     va_end (ap);
     if (!event)
         return -1;
-    if (json_array_append (reslog->eventlog, event) < 0) {
-        json_decref (event);
+    if (!zlistx_add_end (reslog->eventlog, json_incref (event))) {
         errno = ENOMEM;
         return -1;
     }
```

Comment:
if `zlistx_add_end()` fails, I think you gotta call the `json_decref()` b/c you `json_incref()`ed above

Reply:
It might be slightly simpler to just define an `entry_duplicator()`.